Jellyfin.Server.Implementations/Trickplay/TrickplayManager.cs

```diff
@@ -144,14 +204,53 @@ private async Task RefreshTrickplayDataInternal(
                     actualWidth = 2 * ((int)mediaSource.VideoStream.Width / 2);
                 }
 
-                var outputDir = GetTrickplayDirectory(video, actualWidth);
+                var tileWidth = options.TileWidth;
+                var tileHeight = options.TileHeight;
+                var saveWithMedia = libraryOptions is not null ? libraryOptions.SaveTrickplayWithMedia : false;
+                var outputDir = GetTrickplayDirectory(video, tileWidth, tileHeight, actualWidth, saveWithMedia);
 
-                if (!replace && Directory.Exists(outputDir) && (await GetTrickplayResolutions(video.Id).ConfigureAwait(false)).ContainsKey(actualWidth))
+                // Import existing trickplay tiles
+                if (!replace && Directory.Exists(outputDir))
                 {
-                    _logger.LogDebug("Found existing trickplay files for {ItemId}. Exiting", video.Id);
-                    return;
+                    var existingFiles = Directory.GetFiles(outputDir);
+                    if (existingFiles.Length > 0)
+                    {
+                        var existingTrickplayResolution = (await GetTrickplayResolutions(video.Id).ConfigureAwait(false)).ContainsKey(actualWidth);
```

Comment:
i dont like quering the whole dataset just to check for one key. Maybe build a dedicated `Exist` method